RandomMCEPlayer.java

```diff
@@ -0,0 +1,296 @@
+				int play = legalPlayLists[numPlays][j];
+				makePlay(card, play / SIZE, play % SIZE);
+				// System.out.printf("%s: %.1f\n", card, expectiSimPlay());
+				sum += expectiSimPlay()/remainingPlays; // Returns the expected value of this card being played. Divide by remaining plays to scale the returned value by the number of moves that it's testing
+				undoPlay();
+			}
+		}
+		// Calculate expected score by averaging the expected score of all the possible cards that could be pulled
+		return ((float)sum)/(NUM_CARDS-numPlays);
+	}
+
+	/**
+	 * From the chosen play, perform simulated Card draws and random placement (depthLimit) iterations forward
+	 * and return the resulting grid score.
+	 * @param depthLimit - how many simulated random plays to perform
+	 * @return resulting grid score after random MC simulation to given depthLimit
+	 */
+	private int simPlay(int depthLimit) {
+		if (depthLimit == 0) { // with zero depth limit, return current score
+			return system.getScore(grid);
+		}
+		else { // up to the non-zero depth limit or to game end, iteratively make the given number of random plays
+			int score = Integer.MIN_VALUE;
+			int maxScore = Integer.MIN_VALUE;
+			int depth = Math.min(depthLimit, NUM_POS - numPlays); // compute real depth limit, taking into account game end
+			for (int d = 0; d < depth; d++) {
+				// generate a random card draw
+				int c = random.nextInt(NUM_CARDS - numPlays) + numPlays;
+				Card card = simDeck[c];
+				// choose a random play from the legal plays
+
+				int remainingPlays = NUM_POS - numPlays;
+				System.arraycopy(plays, numPlays, legalPlayLists[numPlays], 0, remainingPlays);
+				int c2 = random.nextInt(remainingPlays);
+				int play = legalPlayLists[numPlays][c2];
+				makePlay(card, play / SIZE, play % SIZE);
+			}
+			score = system.getScore(grid);
+
+			// Undo MC plays.
+			for (int d = 0; d < depth; d++) {
+				undoPlay();
+			}
+
+			return score;
+		}
+	}
+
+	public void makePlay(Card card, int row, int col) {
+		// match simDeck to event
+		int cardIndex = numPlays;
+		while (!card.equals(simDeck[cardIndex]))
+			cardIndex++;
+		simDeck[cardIndex] = simDeck[numPlays];
+		simDeck[numPlays] = card;
+
+		// update plays to reflect chosen play in sequence
+		grid[row][col] = card;
+		int play = row * SIZE + col;
+		int j = 0;
+		while (plays[j] != play)
+			j++;
+		plays[j] = plays[numPlays];
+		plays[numPlays] = play;
+
+		// increment the number of plays taken
+		numPlays++;
+	}
+
+	public void undoPlay() { // undo the previous play
+		numPlays--;
+		int play = plays[numPlays];
+		grid[play / SIZE][play % SIZE] = null;
+	}
+
+	/* (non-Javadoc)
+	 * @see PokerSquaresPlayer#setPointSystem(PokerSquaresPointSystem, long)
+	 */
+	@Override
+	public void setPointSystem(PokerSquaresPointSystem system, long millis) {
+		this.system = system;
+	}
+
+	/* (non-Javadoc)
+	 * @see PokerSquaresPlayer#getName()
+	 */
+	@Override
+	public String getName() {
+		return "RandomMCPlayerDepth" + depthLimit;
```

Comment:
We need to update the name here to match with the file name (this is important since the TestRunner uses this `getName()` function to generate the name of the log file for the player).